### ISSUE TITLE
[RFC]: make `axis` kwarg optional in `squeeze`

### ISSUE THREAD

**Issue by ev-br** [data-apis/array-api#892]
Cross-referencing from https://github.com/data-apis/array-api-strict/issues/62 :

The standard's version of `squeeze` has `axis`  as a required argument (https://data-apis.org/array-api/2023.12/API_specification/generated/array_api.squeeze.html#array_api.squeeze), while virtually all array libraries allow `axis=None` with the semantics of "remove all singleton dimensions":

-  `torch.squeeze(input: Tensor, dim: Optional[Union[int, List[int]]])`
- `numpy.squeeze(a, axis=None)`
- `jax.numpy.squeeze(a, axis=None)`

Is there any deep reason to not follow the suit and allow the `axis` to be `Optional[ int | tuple]`?

**Comment by jakevdp** [comment]
Related discussion: #100. In particular, this comment: https://github.com/data-apis/array-api/pull/100#issuecomment-837774492

> I don't think we need "clean up size-1 dimensions if they exist" in the API standard. Squeezing without an axis really only makes sense in an interactive context, but our primary audience here is library developers.

**Comment by ev-br** [comment]
Oh, thank you @jakevdp ! My search-foo failed me on this one, and it seems the subject has been already resolved, quite a while ago at that. 
I'll proceed with bringing array-api-strict in line with the spec then.
